docs/design/configs.md

```diff
@@ -0,0 +1,231 @@
+# Generic Configuration Object
+
+## Abstract
+
+This proposal proposes a generic configuration object named ``ConfigData`` stored inside ``etcd`` that
+stores command-line flags, environment variables, and configuration files used 
```

Comment:
@tmrts Agree with @smarterclayton that I would like to see an example for flags.  That's the one I'm probably most confused about.